### PR TITLE
make parameter_write idempotent

### PR DESCRIPTION
**Pull request by dennybaa**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -1,23 +1,22 @@
...
+  for_each = local.parameter_write
+  name     = each.key
 
-  type            = lookup(tolist(var.parameter_write)[count.index], "type", "SecureString")
-  tier            = lookup(var.parameter_write[count.index], "tier", "Standard")
-  key_id          = lookup(tolist(var.parameter_write)[count.index], "type", "SecureString") == "SecureString" && length(var.kms_arn) > 0 ? var.kms_arn : ""
-  value           = tolist(var.parameter_write)[count.index]["value"]
-  overwrite       = lookup(tolist(var.parameter_write)[count.index], "overwrite", "false")
-  allowed_pattern = lookup(tolist(var.parameter_write)[count.index], "allowed_pattern", "")
+  description     = each.value.description
```
Comment (Gowiem) [inline main.tf:14]: The default for the description previously was the provided name. Did you purposefully remove that from your `parameter_write_defaults` or was that unintentional?

File: versions.tf [hunk 2]
```diff
@@ -1,10 +1,9 @@
 terraform {
-  required_version = ">= 0.12.0, < 0.14.0"
+  required_version = ">= 0.12.6, < 0.14"
```
Comment (Gowiem) [inline versions.tf:2]: Our standard pinning nowadays is to only pin the lower bound. Can you update this and the below `required_providers` block to only pin the `>= XX` value and don't include the upper bounds?

**Review by Gowiem — CHANGES_REQUESTED** [review]
@dennybaa This is looking good. One requested change and a question. 

Also, can you provide the reasoning for this PR in your PR description?

File: main.tf (hunk 1 above)
Comment (dennybaa) [reply main.tf:14]: Well, I think yes. Since there's 0% in the description same as name

**Pull request closed without merging**